xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
Well I got drilled into writing the break conditions into the for statement, so there's a single flow to follow without breaks in the middle that change it. So to me, readability is a lot better with everything in the for statement, than reading 3 times as many lines of code that does the same thing. This is an easy example, but in general, when loops get more complex, breaking in the middle and finding out what you're breaking out of and where you get into decreases readability imo.

And the resulting asm is a bit smaller too if there's a bunch of conditions checked in the single for statement than when you spread it out ;-)